---
title: 'NEED TO CHANGE TITLE'
author: "Kory Melton"
date: "December 7th, 2019"
output: word_document
csl: bioinformatics.csl
bibliography: references.bib
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

```{r load-packages}
# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("lubridate")
library("forcats")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")
```

``` {r add melted phyloseq}
# load in the saved phyloseq object to work with
load("output/phyloseq_obj.Rda")

##########################################
# Make plot ID's more readable
##########################################

phyloseq_obj@sam_data[["plotID"]] <-
  substring(phyloseq_obj@sam_data[["plotID"]],
            6,
            8)

################################################
# Make melted phyloseq
################################################

# make melted phyloseq
# melt phyloseq obj into a data frame for dplyr/ggplot
# analysis and visualization
melted_phyloseq <- psmelt(phyloseq_obj)

# turn all factor columns into character columns for dplyr
melted_phyloseq <- melted_phyloseq %>%
  mutate_if(is.factor, as.character)
```

``` {r add blast data}
# read curated summary data in from csv
blast_results <- read.csv("output/curatedSummary.csv")

# flip the order of the data so the genus comes first
top_10_genus_blast <- blast_results[, c(2, 1)] %>%
  arrange(desc(count)) %>%
  mutate(genus = paste0("*", genus, "*")) %>%
  head(10)

top_10_genus_blast$genus <- gsub("[*]Proteobacteria[*]",
                                 "Proteobacteria",
                                 top_10_genus_blast$genus)

# read curated data on top 5 in from csv
top_5_genus <- read.csv("output/top5blast.csv")

# mutate the data frame to split scientific name into genus and species
top_5_genus$genus <- gsub(" .*$", "", top_5_genus$Scientific.Name)
top_5_genus$species <- gsub(".* ", "", top_5_genus$Scientific.Name)
```
<!-- word count: 385 words -->

# Introduction

<!--Overview of soil bacterial communities (400 words)-->
<!--ACTUAL: 80 words -->

Microscopic organisms are among the oldest and most prevalent lifeforms on earth. Accordingly, they play a vital role in the health of ecosystems by affecting processes like carbon dioxide and nitrogen cycling [@martiny]. As the worlds climate changes over time understanding how microbes influence the dynamics of ecosystems will be necessary to preserve the environment. Currently, the impacts of climate change on the biogeochemical cycling of microbes are not well understood [@barnard]. However, some research suggests microbes may be able to mediate carbon-cycle feedbacks to climate warming [@zhou]. 

Due to their importance in the function and stability of ecosystems, the biodiversity of microorganisms is vital to the overall diversity of macroscopic organisms. Typically the first objective of many ecological research projects is to determine the which organisms are present and their bio-geography [@zimmerman]. Previous research suggests the distribution of microbes is heterogeneous and spatially-structured [@maron]. Thus, accurate sampling of a wide variety of plots is necessary to understanding how microbes influence a large ecosystem. 

For the microbial community, identifying distinct species is particularly difficult [@koeppel]. Recent advances in molecular biology, including high-throughput sequencing has allowed for genetic identification of microbial species without culturing them. Many researchers use amplicon sequencing of the 16S rRNA gene as a bio-marker for the microbial community [@tringe]. 

<!--Overview of NEON -> My site including UNDERC (300 words)-700 -->
<!--ACTUAL: 166 words-->

## UNDE Site

The National Science Foundation (NSF) funded National Ecological Observation Network (NEON) was designed to answer how the ecosystem and it's components respond to changes in climate? [@keller] To this aim NEON collects microbial samples in soil from a number of sites across the United States. I will investigate the UNDE site, near the Great Lakes in Michigan. In the Great Lakes area NEON is particularly interested in land use and forest management. A balanced and well functioning soil microbiome is vital to the health of the forests found in the UNDE site and the Greak Lakes in general.

The UNDE site is managed by the University of Notre Dame Environmental Research Center (UNDERC). The site is at an elevation of 518 m, with a mean annual temperature of 3$^\circ$, and a moderate 854 mms of annual precipitation. The property has a second-growth Northern mesic forest and Evergreen forests. The dominant species are red and sugar maple (*Acer rubrum* and *A. saccharum*), aspen (*Populus tremuloides*), paper birch (*Betula papyrifera*), balsam fir (*Abies balsamea*), cedar (*Thuja occidentalis*), and black spruce (*Picea mariana*). In addition to the deciduous and mixed forests the site has woody wetlands dominated by thicks of alder (*Alnus incana*).

Before the land was donated to the Notre Dame University in the 1930s, region-wide logging for pine cut most of the forested areas on the property. Timber harvest continued into the 1950s, but was followed by a forest regrowth. Recently, the site is used for recreational, educational, and research goals.

The climate is generally humid, cool, and wet with no true dry season and receives approximately 114 inches of snow annually. The soil families include coarse, loamy, mixed, superactive, frigid, argic fragiaquods. Although, some of the soil on the site is poorly drained, which may cause acidic sphagnum bogs. This is especially troublesome since climate warming can promote the growth of peat mosses and compound this issue [@breeuwer2009decreased]. The presence of sphagnum is known to increase the amount of dissolved organic Nitrogen; reducing Nitrogen immobilization by soil microbes [@bragazza2013biogeochemical]. The competition for nutrients and changes caused by sphagnum bogs and climate warming, can shortcircuit the Nitrogen cycle. In years where sphagnum bogs are prevalent one could expect lower abundances of soil microbes. In addition, ecosystems with a high abundance of nitrogen-fixing bacteria may be more suited to withstand the disruption caused by sphagnum bogs.

<!-- Can add more information about this specific site -->

<!-- Do some research into the soil and tree and what microbes should be found -->

<!--Approach (200 words)-900 -->

## Study Objectives

The primary objective of this study is to determine which microbes are present at the UNDE site, using the 16S barcode sequence, and how this changes over time. This information provides insight into how these ecosystems respond to climate change and their overall health and function. The potential for sphagnum bog growth at the UNDE site is a primary concern. To that end, I looked for signs to indicate plots at the site which may have issues with sphagnum growth. A secondary objective of this study is to carefully analyze matches against the BLAST database. Many of my colleagues who have collected data from various NEON sites had *Ralstonia solanacearum* as their top BLAST match, and often had *Staphylococcus aureus* among the top of their list.

The UNDE site has 1192 samples containing sequencing data from soil microbes. The data collected by NEON was be processed for quality control, and sub-sampled. At this point the sub-sampled files will be processed through two pipelines: DADA2 and BLAST. For the BLAST pipeline the samples were trimmed, converted to fasta format and compared to NCBI's nucleotide database using the BLAST algorithm. After comparison to the nucleotide database, I curated the data in R based on percent identity, sequence match length, expected value, and bitscore. For the DADA2 pipeline, I discarded poor sequences and trimmed the higher quality sequences. Then, I genereated sample inferences using error models and compared the results to a training data set. I used the matches from both the nucleotide database and training data set to make inferences on the type of microbial organisms present.  

## Major Findings

<!--Major findings (150 words)-1050 -->

The data set contained a large number of high quality sequences. I found *Ralstonia solanacearum* was the most frequent BLAST match with a few matches with a query cover of 100%. The finding of *Ralstonia solanacearum* is important because this could indicate a soil-borne plant pathogen on the UNDE site.

# Methods
<!--TARGET: 500 words -1550 total -->
<!--ACTUAL: 488 words -->

<!-- 45 words-->
NEON collected and sequenced the samples from the UNDE site, all data can be found in data product DP1.10108.001 (https://data.neonscience.org/data-products/DP1.10108.001) [@NEON]. I analyzed the samples which were formatted and uploaded by NEON to make inferences on the microbial sequences present in each sample over time.

## Sample Collection

<!-- 103 words -->
Each site contains ten plots that are further sampled one to three times per year. NEON randomly selects three locations for each sample event, with each sampling location is not re-sampled. Soil sampling is collected to a maximum depth of thirty centimeters. After collection, samples are stores in sterile containers, frozen and shipped to a lab for downstream analysis. 
In the lab, DNA is extracted and the samples are prepared for high-throughput sequence analysis using primers for 16S sequences. The sequences are delivered to NEON for quality control and acceptance. After acceptance, NEON formats and uploads the files in repositories for public use.

## Analysis

<!-- Overview/general outline of analysis approach -->
<!-- 84 words -->
I downloaded all unique 16S files for the UNDE site from the NEON database. Then, I checked each files quality using fastqc. I sub-sampled each file to make the dataset computationally tractable. I trimmed each file in order to capture high quality reads and converted the fastq files to fasta files. Next, I compared each fasta file to a nucleotide database to make inferences on which microorganisms were present in the samples from the UNDE site. 

Finally, I used those BLAST results to BLANK!!! FILL THIS IN!!!!

### Downloading Data

<!-- 71 words -->
I downloaded all unique files for the target gene 16S from the NEON database for the UNDE site. The script to download the data used the R library neonUtilities and the NEON API to download the data product. I used the data product ID DP1.10108.001. In addition, I downloaded the metadata for the raw data to include information on when the data was collected and where the data was collected from.

### Fastqc

<!-- 28 words -->
I extracted the R1 files from their zipped folders. Then, I ran fastqc on each file to assess their quality and identify any problematic files [@fastqc]. 

### Sub-sampling

<!-- 28 words -->
I randomly sub-sampled 0.5% of the sequences in each file to make the data computationally tractable. For this, I used the srand function and the random seed 1234.

### Trimming

<!-- 54 words -->
I trimmed the sub-sampled files using TrimmomaticSE [@trimmomatic]. I used four threads, phred33 base quality encoding, a minimum quality threshold of 5 for both leading and trailing bases, and a minimum length of two-hundred base pairs. Additionally, I implemented a sliding window with window size eight and required quality of twenty-five.

### BLAST

<!-- 80 words -->
I used bioawk (https://github.com/lh3/bioawk) to convert the sub-sampled, trimmed fastq files to fasta files for downstream BLAST analyses. I compared each fasta file to the GenBank's nucleotide database [@benson2000genbank]. I used the BLASTN algorithm, which uses a nucleotide query sequence to search the nucleotide database [@blastn]. I used four threads, the '10 sscinames std' format, and returned one match for each sequence. I excluded 2018-09-19_environmental_sequence.gi to reduce the number of uncultured or environmental matches. The BLAST results are stored in a comma seperated file in the output directory.

### Analyzing Blast Data

I analyzed the BLAST matches found for each sequence using the seventh bash script in the pipeline. First, I curated the BLAST matches to exclude classifications that were unidentified or too generic. Then, I found the five BLAST matches which occured most frequently throughout the dataset. I analyzed the BLAST matches for the five most prevalent to determine my confidence in the results. 

#### Curating 

I curated the BLAST matches based on the description returned from the BLAST result. I removed all matches with the description uncultured or unidentified since these results would not provide any context on the UNDE site. In addition, I removed all files with the generic names of bacterium, Bacterium, or fungal.

#### Processing Curated File

After curation, I calculated a count for the frequency of each description. Then, I sorted out the top twenty most prevalent matches throughout all files. The count and description are saved in a curated summary comma separated file in the output directory. I also saved an output of the top thirty most prevalent BLAST matches without curation in the output directory.

I saved the data from the BLAST results for the top five most prevalent matches into a comma separated file for further analysis in R. I pulled the ten most prevalent matches based on the highest taxonomic level in the description. This was compared to the top ten genus by abundance from the curated database. Next, I found the most abundant species for each genus.

I used the percent identity, length of sequence match, expected value, and bitscore to assess the quality of the BLAST matches for the top five most prevalent matches by highest taxonomic level. The data were visualized with boxplots using ggplot2 [@ggplot2]. I removed low quality matches based on cutoff values for percent identity, length of sequence match, expected value, and bitscore. I used a cutoff of 90% for percent identity. I sorted out all sequences with a match length below two-hundred base pairs. I used a upper bound of 0.00010 for expected value and two-hundred for bitscore.

### DADA2 Pipeline

I also processed the subsampled data with dada [@dada2]. The sequences were filtered and trimmed to ensure all analyzed sequences were high quality. I discarded any sequences with N's. Only up to three expected errors were allowed. Reads were truncated at the first instance of a quality score below two. I also discarded reads which matched against the phiX genome. Error models were generated for each of the samples. I removed all duplicated sequences. I generated high resolution sample inferences from the dereplicated forward amplicon sequencing reads and the error models previously generated. I removed all chimeras using the consensus method. All sequences shorter than fifty were removed because there is no way to assign their taxonomy. I assigned taxonomy using a supplied training data set while checking the reverse complement of all sequences to see if it is a better match to the reference sequence. The results were saved to the output directory.

### Metadata

I downloaded the metadata for the UNDE site from the DP1.10108.001 data product by NEON. All duplicated rows were removed and matched to the data from the dada2 pipeline. I pruned the samples without any sequences.


### Analysis

I analyzed the data using ggplot [@ggplot2] and dplyr [@dplyr]. First, I calculated the number of samples for each month, year, kingdom, and plot. Then, I found the top phyla in each plot by abundance. I separated the top three phyla by year to analyze the change of the samples from 2016 to 2017. For this analysis I only included the months of May and August since they were the only two months with collections in both 2016 and 2017. I analyzed the the abundance of Phyla by month and year to determine if the composition of samples changed over time. Then, I took the top six genus for each month and year separated by Phyla. Finally, I used [@phyloseq] to visualize the alpha diversity by month and year using the shannon method.

# Results
<!-- 600 words 2150 total -->
<!-- Add error plots -->

<!-- OVERVIEW OF RESULTS -->

## Sequence Counts and Quality

The data set included 1192 fastq files with a range of sequences per file. The maximum number of sequences in a file was 413,851 sequences, while the minimum number of sequences was 118. The average number of sequences in a file was 43,028 sequences. There was a total of 51,289,602 sequences in the data set.

After sub-sampling there was 268,512 sequences total, or approximately .5% of the sequences. The maximum number of sequences after sub-sampling was 2,090, while the minimum was zero. Seven of the files did not have any sequences after sub-sampling. This was about 0.6% of the raw fastq files. 

After trimming there was 154,922 sequences total, or approximately 58% of the sub-sampled sequences. The maximum number of sequences after trimming was 1,213, while the minimum was zero. Nineteen of the files did not have any sequences after sub-sampling. This was about 1.6% of the raw fastq files. files.

Overall, the sequence quality was good. None of the inspected fastqc reports contained sequences flagged as poor quality. Figure 1 shows the fastqc of a file which did not have any sequences after trimming, while Figure 2 shows the fastqc of the file with the most sequences after trimming. The decline in quality scores happens about 50 base pairs later in the better file.

## BLAST Results

The four most frequent BLAST matches I found in the data set were *Ralstonia solanacearum*, *Staphylococcus aureus*, *Geobacillus thermocatenulatus*, and *Sphingomonas melonis*.

The top matches I found in each file with a query cover of 100% were *Ralstonia solanacearum*, *Bradyrhizobium elkanii*, *Bacillus sp.*, *Actinomadura sp*, and *Acetobacteraceae bacterium*. 

The ten most frequent BLAST matches by the highest taxonomic level can be seen in Table 1. They are *Ralstonia*, *Staphylococcus*, *Bradyrhizobium*, *Pseudomonas*, Proteobacteria, *Candidatus*, *Mucilaginibacter*, *Bacillus*, *Sphingomonas*, and *Burkholderia*. *Ralstonia* dominated the BLAST matches with 68,508 total matches in the data set. There was a large dropoff to *Staphylococcus* and *Bradyrhizobium*.

The ten most frequent genus matches by abundance were *Bradyrhizobium*, *Pseudomons*, *Roseiarcus*, *Burkholderia*, *Mycobacterium*, *Thermosporothrix*, *Gaiella*, *Massilia*, *Pedomicrobium*, and *Gemmatimonas*. *Bradyrhizobium*, *Pseudomonas*, and *Burkholderia* were prevalent in both lists while *Bradyrhizobium* and *Pseudomonas* appeared at near the top of the NCBI search and the curated search.

The species returned for each genus in the BLAST results can be seen in Figure 3 and Table 3. For *Ralstonia* and *Staphylococcus* one species dominated all of the BLAST matches for the genus. *Ralstonia solanacearum* was present in 68,505 of the 68,508 of the *Ralstonia* matches, or 99.99% of the matches. *Staphylococcus aureus* was present in 3,059 of the 3,068 *Staphylococcus* matches, or 99.71% of the matches. The BLAST matches for *Bradyrhizobium* and *Pseudomonas* had a wider variety of species. Figure 3 shows the the top three species for these two genera. *Bradyrhizobium* was dominated by *Bradyrhizobium elkanii* and other unidentified species with a small number of *Bradyrhizboium erythroplei*. *Pseudomonas* had a lot fewer samples overall, but they were equally spread among *Pseudomonas frederiksbergensis*, *Pseudomonas protegens*, and unidentified species. 

Figues 4 through 7 show the distribution of values for the BLAST matches by their description. In each graph there is a red line depicting the cutoff value used for the corresponding value. As shown in Figure 4, a large majoriy of the percent identities for the BLAST matches fall close to 100%. However, all matches except for proteobacteria have matches with lower than 90% query cover. Figure 5 highlights the difference of the length of sequences by the different types of BLAST matches. A majoriy of the matches for *Ralstonia* and *Staphylococcus* are fairly short sequences less than 100 base pairs long. While proteobacteria, *Bradyrhizobium*, and *Pseudomonas* had a majority of their matches above 200 base pairs long. In addition, as seen in Figure 6, *Ralstonia* and *Staphylococcus* have higher expected values with many above the cutoff of 0.00010. *Bradyrhizobium*, proteobacteria, and *Pseudomonas* have low expected values which indicate strong confidence in the BLAST matches. A similar trend can be seen with the bitscore values returned from the BLAST matches. In figure 7, you can see *Ralstonia* and *Staphyloccocus* have low bitscores centering around 100 base pairs while *Bradyrhizobium*, proteobacteria, and *Pseudomonas* had bitscores centering around 500. Table 4 shows the BLAST results before and after curation to throw out lower quality matches. The counts for *Ralstonia* and *Staphyloccocus* drop drastically with curation while a majority of the matches to *Bradyrhizobium*, proteobacteria, and *Pseudomonas* remain after curation. With all of these metrics considered, I have high confidence of the presence of *Bradyrhizobium*, *Pseudomonas*, and proteobacteria in the samples from the UNDE site. In contrast, I am not confident there was actually *Ralstonia* and *Staphylococcus* present at the UNDE site.

## Samples over Time

The number of samples over varied quite a lot between the time of collection and the plot collected from. Figure 8 shows the variation of the number of samples collected by month and year. Only May and July had samples collected in both 2016 and 2017. In both of these months there was significantly more samples collected in 2016 than 2017. In addition, the most samples were collected in May; followed by August, October, and finally July. Figure 9 shows the samples by plot in 2016 and 2017 during the months of May and August. In total there was eighteen plots sampled from the UNDE site. Seven of the eighteen plots were sampled in both 2016 and 2017: 001, 006, 008, 037, 038, 043, 044. Eight of the plots were only sampled in 2016: 007, 013, 014, 017, 019, 027, 034, and 035. Three of the plots were only sampled in 2017: 002, 003, and 010. Both archaea and bacteria were present in 2016 and 2017. There was 22  distinct archaea samples and 1,757 distinct bacterial samples matched to the curated database. The dataset contained 1,099 distinct samples in 2016. In contrast, the dataset contained 710 samples in 2017. The relative number of samples by year and Kingdom is illustrated in Figure 10. 

Next, I analyzed the abundance of different Phyla by plot, year, and month. For sixteen of the eighteen plots proteobacteria was the most abundant Phylum. In plot 002 acidobacteria was the most abundant Phylum. Verrucomicrobia was the most abundant Phylum in plot 014. In both cases this abundance was much lower than many of the plots with Proteobacteria as the most abundant Phylum. For both years, Proteobacteria, Acidobacteria, and Actinobacteria were the three most abundant Phyla. The ratio of Phyla remained constant from 2016 to 2017. However, the abundance for each Phyla significantly declined from 2016 to 2017. Unsurprisingly, Proteobacteria was the most abundant Phyla in each month collected for both years. The abundances for all of the Phyla were significantly lower in 2017. Interestingly, no Actinobacteria were found in may of 2017 while it was the third most abundant Phyla in 2016. No samples were collected in October of 2016 and July of 2017.

Since proteobacteria was the most abundant Phyla I looked into the abundance of different genera of Proteobacteria. The most abundant proteobacteria was *Bradyrhizobium* followed by *Pseudomonas*, *Roseiarcus*, and *Burkholderia*. Proteobacteria as a whole were more abundant during May and August. However, *Pseudomonas* and *Roseiarcus* were almost twice as abundant during May than August. *Gaiella* were the most abundant Actinobacteria, followed by *Myobacterium*. *Gaiella* was only present during May.

Finally, I assessed the diversity of samples by plot by month and by year. Diversity was much higher for each plot during 2016. However, there are a significantly fewer diversity points plotted for 2017. For the months May, July, and August there does not appear to be a clear pattern for diversity. In plots 001, 007, 027, 035, 037, and 038 diversity seems to be higher in August than May. In plot 019 diversity appears to be higher in May than July. The highest diversity was in plot 001 in August.


<!-- talk about the variation of plots if I need to add words -->
<!-- add more sections and subsections-->

# Discussion
<!-- 500 words 2650 total -->

## Overview

The UNDE site, in Michigan, near the great lakes has a second-growth Northern mesic forest and Evergreen forests with various trees including maple, aspen, birch, cedar, and black spruce. In addition, the site has woody wetlands dominated by thicks of alder. 

The data collected by NEON from this site was high-quality and contained a substantial amount of genetic information. The collected sequences matched a variety of bacteria widely associated to soil including *Ralstonia solanacearum* and *Acetobacteraceae bacterium*. These two species have very different impacts on plants. *Ralstonia solanacearum* is a plant pathogen while *Acetobacteraceae bacterium* can be found in symbiotic relationships with many different plants by colonizing their internal tissues.

## BLAST Interpretation

*Ralstonia solanacearumm* is a soil-borne plant pathogen with unusually high prevalence worldwide [@salanoubat2002genome]. This pathogen naturally infects the roots of plants causing bacterial wilt in many plants and crops [@castillo2007evolutionary]. The DNA sequencing of the soil samples from the UNDE site indicate a high probability of *Ralstonia solanacearum*. The workers of UNDERC should look for indications of this pathogen in plants near the sampling sites and take preventative action to protect the biodiversity of the UNDE site.

## Taxonomic Interpretation

## Limitations

## Future Work


# Tables and Graphs

![](data/images/fastqc_scores/BMI_Plate1WellA1_16S_R1_fastqc_scores.png)

**Figure 1.** Fastqc per base sequence quality for BMI_Plate1WellA1_16S_R1. This was one of the lower quality files and did not have any acceptable sequences after sub-sampling and trimming. You can see a sharp decline on base sequence quality around 210 base pairs.

![](data/images/fastqc_scores/BMI_Tube52_16S_R1_fastqc.png)

**Figure 2.** Fastqc per base sequence quality for BMI_Tube52_16S_R1. This was one of the higher quality files. You can't see a sharp decline until approximately 260 base pairs.

``` {r blast table}
# make table
kable(top_10_genus_blast,
      col.names = c("Description", "Count"))
```

**Table 1.** Top 10 genus returned from blast results

``` {r blast comparison}
# get top 20 genus
top_10_genus_curated <- melted_phyloseq %>%
  group_by(Genus) %>%
    filter(!is.na(Genus)) %>%
      summarize(sum_abund = sum(Abundance,
                                na.rm = TRUE)) %>%
        arrange(desc(sum_abund)) %>%
          mutate(Genus = paste0("*", Genus, "*")) %>%
            head(10)

kable(top_10_genus_curated,
      col.names = c("Genus", "Abundance"))
```

**Table 2.** Top 10 genus by abundance from curated database

``` {r detailed blast}
# get top by species w/o Bradyrhizobium and Pseudomonas
top_3_staph_rals_pro <- top_5_genus %>%
  filter(genus != "Bradyrhizobium") %>%
  filter(genus != "Pseudomonas") %>%
  group_by(Scientific.Name) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(3)

# change variable to match snake case
colnames(top_3_staph_rals_pro)[1] <- "scientific_name"

top_3_staph_rals_pro$scientific_name <-
  paste0("*",
         top_3_staph_rals_pro$scientific_name,
         "*")
top_3_staph_rals_pro$scientific_name <-
  gsub("[*]Proteobacteria[*]",
       "Proteobacteria",
       top_3_staph_rals_pro$scientific_name)
  
# get top 3 of Bradyrhizobium and Pseudomonas
# Bradyrhizobium
top_3_br <- top_5_genus %>%
  filter(genus == "Bradyrhizobium") %>%
  group_by(genus, species) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(3)

# Pseudomonas
top_3_pseu <- top_5_genus %>%
  filter(genus == "Pseudomonas") %>%
  group_by(genus, species) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(3)

# bind Bradyrhizobium and Pseudomonas
top_3_bp <- rbind(top_3_br,
                  top_3_pseu)

# graph counts with genus on x and fill species
# Filtered for only the top 10 samples
# Ralstonia was ommitted to make graph more readable
# Proteobacteria didn't have any species
top_3_bp %>%
  filter(n > 10) %>%
  ggplot(aes(x = genus,
             y = n,
             fill = species)) +
  xlab("Genus") +
  ylab("Number of Samples") +
  ggtitle("BLAST Samples by Genus and Species") +
  geom_col(position = position_dodge()) +
  theme(axis.text.x = element_text(angle = 60,
                                   hjust = 1))
```

**Figure 3.**

``` {r detailed-table}
# make a table for other relevant counts
kable(top_3_staph_rals_pro,
      col.names = c("Description",
                    "Count"))
  
```

**Table 3.**

``` {r percent Identity}

# graph boxplot of percent identity by top 5 Genus
top_5_genus %>%
  ggplot(aes(x = genus,
             y = pident)) +
  geom_boxplot() +
  ggtitle("Percent Identity by Description") +
  xlab("Description") +
  ylab("Percent Identity") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 60,
                                   hjust = 1)) +
  geom_hline(yintercept = 90, color = "red")
```

**Figure 4.**

``` {r length}
# graph boxplot of length by top 5 Genus
top_5_genus %>%
  ggplot(aes(x = genus,
             y = length)) +
  geom_boxplot() +
  ggtitle("Length of Sequence Match by Description") +
  xlab("Description") +
  ylab("Length of Sequence") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 60,
                                   hjust = 1)) +
  geom_hline(yintercept = 200, color = "red")
```

**Figure 5.**

``` {r evalue}
# graph boxplot of evalue by top 5 genus
top_5_genus %>%
  filter(evalue < .0002) %>%
  ggplot(aes(x = genus,
             y = evalue)) +
  geom_boxplot() +
  ggtitle("Expected Value by Description") +
  xlab("Description") +
  ylab("Expected Value") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 60,
                                   hjust = 1)) +
  geom_hline(yintercept = 0.0001, color = "red")
```

**Figure 6.**

``` {r bitscore}
# graph boxplot of bitscore by top 5 genus
top_5_genus %>%
  ggplot(aes(x = genus,
             y = bitscore)) +
  geom_boxplot() +
  ggtitle("Bitscore by Description") +
  xlab("Description") +
  ylab("Bitscore") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 60,
                                   hjust = 1)) +
  geom_hline(yintercept = 200, color = "red")
```

**Figure 7.**

``` {r curated v. uncurated, message=FALSE}
# Number of Samples in top 5 genus when controlled for
# - bitscore above 200
# - e value below 0.05
# - length above 200
# - percent identity above 90%
curated_top_5_totals <- top_5_genus %>%
  filter(bitscore > 200) %>%
  filter(evalue < 0.0001) %>%
  filter(length > 200) %>%
  filter(pident > 90) %>%
  group_by(genus) %>%
  tally()

top_5_totals <- top_10_genus_blast %>%
  head(5) %>%
  arrange(genus)

invisible(sort(top_5_totals$genus))

top_5_totals$curated_count <- curated_top_5_totals$n

top_5_totals <- top_5_totals %>%
  arrange(desc(count))

kable(top_5_totals,
      col.names = c("Description",
                    "Count",
                    "Curated Count"))
```

**Table 4.**

``` {r Samples by year}
##########################################
# Number of Samples
##########################################

# Get number of samples by year, plot, month
sample_info <- melted_phyloseq %>%
  distinct(OTU, .keep_all = TRUE) %>%
  group_by(plotID, collect_year, collect_month) %>%
  tally()

# Graph the number of samples by month, with fill year
sample_info %>%
  ggplot(aes(x = factor(collect_month),
             y = n,
             fill = factor(collect_year))) +
  xlab("Collection Month") +
  ylab("Number of Samples") +
  geom_col() +
  labs(fill = "Collection Year") +
  ggtitle("Number of Samples by Month")
```

**Figure 8.** Number of samples by month

``` {r samples by plot}
# Graph the plots in month 5 and 8
sample_info %>%
  filter(collect_month == 5 | collect_month == 8) %>%
  group_by(plotID, collect_year) %>%
  arrange(desc(n)) %>%
  ggplot(aes(x = plotID,
             y = n,
             fill = factor(collect_year))) +
  xlab("Plot ID") +
  ylab("Number of Samples") +
  ggtitle("Samples by plot in May and August") +
  geom_col() +
  theme(axis.text.x = element_text(angle = 60,
                                   hjust = 1)) +
  theme(legend.position = "top") +
  labs(fill = "Collection Year")
```

**Figure 9.** Number of samples by plot

``` {r kingdom analysis}
kingdom_info <- melted_phyloseq %>%
  filter(!is.na(Kingdom)) %>%
  distinct(OTU, .keep_all = TRUE) %>%
  group_by(Kingdom, collect_year, collect_month) %>%
  tally()

kingdom_info %>%
  ggplot(aes(x = factor(collect_year),
             y = n,
             fill = Kingdom)) +
  geom_col() +
  ggtitle("Kingdom by Year") +
  xlab("Collection Year") +
  ylab("Number of Samples")
```
  
**Figure 10.** Summary of Kingdom found matched in phyloseq. 

``` {r phyla by plot}
# list the top phyla in descending order
phyla_by_plot <- melted_phyloseq %>%
  group_by(plotID, Phylum) %>%
  filter(!is.na(Phylum)) %>%
  summarize(sum_abund = sum(Abundance,
                            na.rm = TRUE)) %>%
  filter(sum_abund > 0) %>%
  arrange(desc(sum_abund))

# keep only the top phyla
top_phyla_by_plot <- phyla_by_plot[!duplicated(phyla_by_plot$plotID), ]

top_phyla_by_plot %>%
  ggplot(aes(x = plotID,
             y = sum_abund,
             fill = Phylum)) +
  geom_col() +
  theme(axis.text.x = element_text(angle = 10)) +
  theme(legend.position = "top") +
  ggtitle("Top Phyla in each plot by Abundance") +
  xlab("Plot ID") +
  ylab("Abundance")
```

**Figure 11.**

``` {r top 3 phyla by year}
# Phyla by year
# control for
# - month 5 or month 8
# - plots: 001, 006, 008, 037, 038, 043, 044
# look at different phylum abundances

# list of plots to keep
plots <- c("001", "006", "008", "037", "038", "043", "044")

# counts of the top 3 phyla overall, for each of the plots
top_3_diverse_phyla <- melted_phyloseq %>%
  filter(Abundance > 0, !is.na(Phylum)) %>%
  distinct(OTU, .keep_all = TRUE) %>%
  group_by(Phylum) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(3) %>%
  pull(Phylum)

melted_phyloseq %>%
  filter(collect_month == 5 | collect_month == 8) %>%
  filter(plotID %in% plots) %>%
  group_by(collect_year, Phylum) %>%
  summarize(sum_abund = sum(Abundance,
                            na.rm = TRUE)) %>%
  filter(Phylum %in% top_3_diverse_phyla) %>%
  ggplot(aes(x = factor(collect_year),
             y = sum_abund,
             fill = Phylum)) +
  xlab("Collection Year") +
  ylab("Abundance") +
  geom_col(position = position_dodge()) +
  ggtitle("Top 3 Phyla by Year")
```

**Figure 12.**

``` {r phyla by month and year}
# phyla by month and year
diverse_phyla <- melted_phyloseq %>%
  filter(Abundance > 0, !is.na(Phylum)) %>%
  distinct(OTU, .keep_all = TRUE) %>%
  group_by(Phylum) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(5) %>%
  pull(Phylum)

melted_phyloseq %>%
  group_by(collect_month, collect_year, Phylum) %>%
  summarize(sum_abund = sum(Abundance,
                            na.rm = TRUE)) %>%
  filter(Phylum %in% diverse_phyla) %>%
  ggplot(aes(x = Phylum,
             y = sum_abund,
             fill = Phylum)) +
  xlab("") +
  ylab("Abundance") +
  geom_col(position = position_dodge()) +
  facet_grid(collect_year ~ collect_month) +
  theme(axis.text.x = element_blank()) +
  ggtitle("Abundance of Phyla by Month and Year")
```

**Figure 13.**

``` {r genus by month and phyla}
diverse_genus <- melted_phyloseq %>%
  filter(collect_year == 2016) %>%
  filter(Abundance > 0, !is.na(Genus)) %>%
  distinct(OTU, .keep_all = TRUE) %>%
  group_by(Genus) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(6) %>%
  pull(Genus)

melted_phyloseq %>%
  filter(collect_month != 10) %>%
  group_by(collect_month, Genus, Phylum) %>%
  summarize(sum_abund = sum(Abundance,
                            na.rm = TRUE)) %>%
  filter(Genus %in% diverse_genus) %>%
  ggplot(aes(x = Genus,
             y = sum_abund,
             fill = Genus)) +
  xlab("") +
  ylab("Abundance") +
  geom_col(position = position_dodge()) +
  facet_grid(Phylum ~ collect_month) +
  theme(axis.text.x = element_blank()) +
  ggtitle("Abundance of Genus in Proteobacteria by Month")
```

``` {r richness by year}
phyloseq_5_8 <- prune_samples(
  phyloseq_obj@sam_data$collect_month == 5 |
  phyloseq_obj@sam_data$collect_month == 8,
                              phyloseq_obj)

# richness in months 5 and 8 with year as the color
plot_richness(phyloseq_5_8,
              x = "plotID",
              measures = "Shannon") +
  xlab("Plot ID") +
  geom_point(aes(color = factor(collect_year))) +
  labs(color = "Year")
```

**Figure 14.**

``` {r richness 2016}
phyloseq_2016 <- prune_samples(
  phyloseq_obj@sam_data$collect_year == 2016,
                               phyloseq_obj)
# richness in 2016
plot_richness(phyloseq_2016,
              x = "plotID",
              measures = "Shannon") +
  xlab("Plot ID") +
    geom_point(size = 2,
               aes(color = factor(collect_month))) +
  labs(color = "Month")
```

**Figure 15.**

# Sources Cited
