# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("lubridate")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")

# this is to get the NEON metadata and can be installed from CRAN
library("neonUtilities")

# Download the full set of metadata for the site over all time periods
metadata_all <- loadByProduct("DP1.10108.001",
                              site = "UNDE",
                              startdate = NA,
                              enddate = NA,
                              package = "expanded",
                              avg = "all",
                              check.size = FALSE)

# pull out the one table we want from the list object
metadata_subset <- metadata_all$mmg_soilMarkerGeneSequencing_16S

# use lubridate functions to parse out the year, month, and day
# of sampling and put these into their own columns in the metadata
metadata_subset <- metadata_subset %>%
  mutate(collect_year = year(parse_date_time(collectDate,
                                             orders = "YmdHM")),
         collect_month = month(parse_date_time(collectDate,
                                             orders = "YmdHM")),
         collect_day = day(parse_date_time(collectDate,
                                             orders = "YmdHM")))

# get rid of any duplicate rows
metadata_subset <- metadata_subset %>%
  distinct(internalLabID, .keep_all = TRUE)

# set the rownames of the metadata to match the sample names used for
# the sample data
rownames(metadata_subset) <- metadata_subset$internalLabID

# load in data files generated by dada
# save necessary files from dada pipeline to use with phyloseq
load("output/dada-results/seqtable.Rda")
load("output/dada-results/taxatable.Rda")

# trim rownames to match the internal lab ID
# rownames(sequence_table_nochim) <- gsub(pattern = "_16S_R1_subsampled_filt",
#                                         replacement = "",
#                                         rownames())

# subset the sequence table to only include those that are also in the
# metadata table
sequence_table_nochim <- sequence_table_nochim[
  rownames(sequence_table_nochim) %in%
    as.character(metadata_subset$internalLabID), ]

# Construct phyloseq object (straightforward from dada2 outputs)
phyloseq_obj <- phyloseq(otu_table(sequence_table_nochim,
                                   taxa_are_rows = FALSE), # sample-spp matrix
                         sample_data(metadata_subset), # metadata for samples
                         tax_table(taxa)) # taxonomy for each sequence variant

# remove the samples without any sequences
phyloseq_obj <- prune_samples(sample_sums(phyloseq_obj) > 0, phyloseq_obj)

# save phyloseq object to use in the Rmd file
save(phyloseq_obj, file = "output/phyloseq_obj.Rda")